command/host_create.go

```diff
@@ -61,8 +67,103 @@ func (c *createHost) Execute(ctx context.Context, comm client.Communicator,
 			return err
 		}
 	}
+	startTime := time.Now()
+	ids, err := comm.CreateHost(ctx, taskData, *c.CreateHost)
+	if err != nil {
+		return errors.Wrap(err, "error creating host")
+	} else if c.CreateHost.CloudProvider == apimodels.ProviderDocker {
+		if err = c.getLogsFromNewDockerHost(ctx, logger, comm, ids, startTime); err != nil {
+			return errors.Wrap(err, "problem getting logs from created host")
+		}
+	}
+
+	return nil
+}
+
+func (c *createHost) getLogsFromNewDockerHost(ctx context.Context, logger client.LoggerProducer, comm client.Communicator,
+	ids []string, startTime time.Time) error {
+	var err error
+	if len(ids) == 0 {
+		return errors.New("Programmer error: no intent host ID received")
+	}
+	if c.CreateHost.StderrFile == "" {
+		c.CreateHost.StderrFile = fmt.Sprintf("%s.err.log", ids[0])
+	}
+	if c.CreateHost.StdoutFile == "" {
+		c.CreateHost.StdoutFile = fmt.Sprintf("%s.out.log", ids[0])
+	}
+
+	if !c.CreateHost.Background {
+		return errors.Wrap(c.waitForLogs(ctx, comm, logger, startTime, ids[0]), "error waiting for logs")
+	}
+
+	go func() {
+		if err = c.waitForLogs(ctx, comm, logger, startTime, ids[0]); err != nil {
+			logger.Task().Errorf("error waiting for logs in background: %v", err)
+		}
+	}()
+
+	return nil
+}
+
+func (c *createHost) waitForLogs(ctx context.Context, comm client.Communicator, logger client.LoggerProducer,
+	startTime time.Time, hostID string) error {
+	pollTicker := time.NewTicker(time.Duration(c.CreateHost.PollFrequency) * time.Second)
+	defer pollTicker.Stop()
+
+	timeoutTimer := time.NewTimer(time.Duration(c.CreateHost.ContainerWaitTimeoutSecs) * time.Second)
+	defer timeoutTimer.Stop()
+
+	batchStart := startTime
```

Comment:
Is the route implemented in a way such that we won't get duplicated messages if one is logged exactly on the time that we send here?